docs/user_guide/02_medrecord_intro.md

```diff
@@ -0,0 +1,239 @@
+# MedRecord
+
+## Preface
+
+Just like every major library has its core class — [PyTorch](https://pytorch.org/) has `torch.Tensor`, [Numpy](https://numpy.org/) has `np.array` — MedModels revolves around `mm.MedRecord` as its central object.
+
+MedModels delivers advanced data analytics methods out-of-the-box by utilizing a structured approach to data storage. This is enabled by the MedRecord class, which organizes data of any complexity within a graph structure. With its Rust backend implementation, MedRecord guarantees high performance, even when working with extremely large datasets.
+
+
+```{literalinclude} scripts/02_medrecord_intro.py
+---
+language: python
+lines: 3
+---
+```
+
+## Adding Nodes to a MedRecord
+
+Let's begin by introducing some sample medical data:
+
+:::{list-table} Patients
+:widths: 15 15 15 15
+:header-rows: 1
+
+*   - ID
+    - Age
+    - Sex
+    - Loc
+*   - Patient 01
+    - 72
+    - M
+    - USA
+*   - Patient 02
+    - 74
+    - M
+    - USA
+*   - Patient 03
+    - 64
+    - F
+    - GER
+:::
+
+This data, stored for example in a Pandas DataFrame, looks like this:
+
+```{literalinclude} scripts/02_medrecord_intro.py
+---
+language: python
+lines: 5-13
+---
+```
+
+To work with this data in MedModels, we'll have to create a MedRecord. The MedRecord employes a builder pattern to easily construct complex datastructures. 
+The Builder Pattern simplifies creating complex objects by constructing them step by step. It improves flexibility, readability, and consistency, making it easier to manage and configure objects in a controlled way.
```

Comment:
this does not explain how the code below works.

IMO it should say something like:

> In the example below we create a new medrecord using the builder pattern. We create a `MedRecordBuilder` and tell it to add the pandas dataframe as nodes with the column "ID" being the column we'll use for indexing. We also tell it to add these nodes into the group "Patients". ...

Reply:
we should also explain somewhere (not in the medrecord module) that the builder pattern is an essential part of the library and can be used to create all major classes. (Not in this PR, just a thought)

Reply:
@philippgrosser I agree but what I wanted to convey is that I think we need a general section somewhere (maybe just the user guide index page) that is about the similarities between all classes (every major class has a builder pattern that works in the same way, every data in-outuput is a medrecord, etc.)